document/core/valid/instructions.rst

```diff
@@ -1491,6 +1924,84 @@ Control Instructions
    }
 
 
+.. _valid-br_on_cast:
+
+:math:`\BRONCAST~l~\X{rt}_1~\X{rt}_2`
```

Comment:
Do all of the below match the decisions made (or pending) in https://github.com/WebAssembly/gc/issues/381?

Reply:
This PR reflects what's currently in MVP.md. Updating it is a separate action item blocked on #381.